src/convert.jl

```diff
@@ -20,7 +20,11 @@ end
 @pure has_size(::Type{<:StaticArray{S}}) where {S<:Tuple} = @isdefined S
 @pure has_size(::Type{<:StaticArray}) = false
 # workaround for https://github.com/JuliaArrays/StaticArrays.jl/issues/1047
-@pure has_size(::Type{SVector}) = false
+for T in (:S, :M)
+    @eval has_size(::Type{$(Symbol(T,:Vector))}) = false
+    @eval has_size(::Type{$(Symbol(T,:Matrix))}) = false
+    @eval has_size(::Type{$(Symbol(T,:Matrix)){N}}) where {N} = false
+ end
```

Comment:
```suggestion
for T in (:S, :M)
    @eval has_size(::Type{$(Symbol(T,:Vector))}) = false
    @eval has_size(::Type{$(Symbol(T,:Matrix))}) = false
    @eval has_size(::Type{$(Symbol(T,:Matrix)){N}}) where {N} = false
end
```

Alternatively, I think we can remove the for loop.

```suggestion
has_size(::Type{SVector}) = false
has_size(::Type{MVector}) = false
has_size(::Type{SMatrix}) = false
has_size(::Type{MMatrix}) = false
has_size(::Type{SMatrix{N}}) where {N} = false
has_size(::Type{MMatrix{N}}) where {N} = false
```